src/Native/Runtime/amd64/AllocFast.asm

```diff
@@ -101,6 +101,56 @@ NewOutOfMemory:
         jmp         RhExceptionHandling_FailedAllocation
 NESTED_END RhpNewObject, _TEXT
 
+;; Allocate one dimensional, zero based array (SZARRAY).
+;;  RCX == EEType
+;;  EDX == element count
+LEAF_ENTRY RhNewString, _TEXT
+
+        ; we want to limit the element count to the non-negative 32-bit int range
+        cmp         rdx, 07fffffffh
```

Comment:
I think that this depends on invariant that `limit - ptr < LARGE_OBJECT_SIZE`, so the fast does not need to check for the large object size explicitly.